WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationDetailFragment.kt

```diff
@@ -165,11 +166,16 @@ class VariationDetailFragment : BaseFragment(), BackPressListener, NavigationRes
             )
         }
         handleResult<List<Image>>(BaseProductEditorFragment.KEY_IMAGES_DIALOG_RESULT) {
-            if (it.isNotEmpty()) {
-                viewModel.onVariationChanged(
-                    image = it.first()
-                )
+            val updatedImage = if (it.isEmpty()) {
+                // Image was deleted. Create a placeholder image with ID 0
+                Image(0, "", "", Date())
+            } else {
+                it.first()
             }
+
+            viewModel.onVariationChanged(
+                image = updatedImage
+            )
```

Comment:
np: if you think it's useful, it's possible to make this expression simpler taking advantage of the `firstOrNull` expression.
```suggestion
            viewModel.onVariationChanged(
                // If null, the image was deleted. Create a placeholder image with ID 0
                image = it.firstOrNull()
                    ?: Image(0, "", "", Date())
            )
```

Reply:
Ooh, great suggestion! Implemented in 77fb7d609301e9b2739909ab1151ae7256a5d515